### ISSUE TITLE
Removing required js files as well

### ISSUE THREAD

**Issue by DarkSouL11** [fqborges/webpack-fix-style-only-entries#23]
Removing one the required js files as well only on production build.
Entry is configures as
```js
entry: {
  styles: path.resolve(commonDir, 'index.scss'),
  main: path.resolve(appsDir, 'main', 'index.js'),
  app1: path.resolve(appsDir, 'app1', 'index.js')
}
```
Removes the generated `main.js` file.
This the content of `index.js`file pointed by main entry.
```js
import React from 'react';
import ReactDOM from 'react-dom';

import App from './App';

import './index.scss';

ReactDOM.render(<App />, document.getElementById('root'));

if (process.env.NODE_ENV === 'development') {
  module.hot.accept();
}

```

**Comment by fqborges** [comment]
Could you give me a [Minimal, Reproducible Example](https://stackoverflow.com/help/minimal-reproducible-example).

**Comment by braedon** [comment]
I believe I've run into this issue as well (but can open a separate issue if needed).

There seems to be two issues at play.

Firstly, in some situations not all resources for a JavaScript entry are collected - namely the actual entry resource itself is missing from the list (others may be as well - unclear). Unless the entry has some other non-style resources that are successfully collected, it will be incorrectly classified as style only and removed.

Compounding this is the second issue. The `_collectedModules` array is global, so shared between all the entries. Shared resources are therefore excluded from the resources collected for later entries, as they're already in `_collectedModules`. This increases the likelihood a given entry will end up having all style resources.

I don't know enough about webpack to work out why the first issue is occurring, but I eventually managed to reproduce it with a (relatively) minimal example - I'll post it in a separate comment.

The second issue seems like an easy fix, however - make `_collectedModules` entry-specific (it seems like it should be, if I'm understanding the code correctly).

**Comment by braedon** [comment]
Here's an example of the issue, as minimal as I could make it. I'm using Svelte, not React, as that's what the project I ran into the issue on uses.

[example.zip](https://github.com/fqborges/webpack-fix-style-only-entries/files/4474759/example.zip)

Project Layout:
```
/example$ tree
.
├── package.json
├── src
│   ├── bar.js
│   ├── components
│   │   ├── bar.svelte
│   │   └── foo.svelte
│   └── foo.js
└── webpack.config.js

2 directories, 6 files
```
`foo.js` and `bar.js` are the two entrypoints, and they each import and run their corresponding component. This seemed to be critical to triggering the first issue.

To reproduce, install and build-prod:
```
/example$ npm install
/example$ npm run build-prod

> Example@0.1.0 build-prod /example
> webpack -p --mode production

webpack-fix-style-only-entries: removing js from style only module: foo.js
Hash: 1deef50477048fff0ba1
Version: webpack 4.42.1
Time: 1193ms
Built at: 14/04/2020 6:41:43 pm
  Asset      Size  Chunks             Chunk Names
bar.css  29 bytes       0  [emitted]  bar
 bar.js  4.63 KiB       0  [emitted]  bar
foo.css  29 bytes       1  [emitted]  foo
Entrypoint foo = foo.css
Entrypoint bar = bar.css bar.js
[3] ./src/foo.js + 1 modules 853 bytes {1} [built]
    | ./src/foo.js 104 bytes [built]
    |     + 1 hidden module
[4] ./src/bar.js + 1 modules 853 bytes {0} [built]
    | ./src/bar.js 104 bytes [built]
    |     + 1 hidden module
    + 5 hidden modules
Child mini-css-extract-plugin node_modules/css-loader/dist/cjs.js!src/components/bar.svelte.css:
    Entrypoint mini-css-extract-plugin = *
       2 modules
Child mini-css-extract-plugin node_modules/css-loader/dist/cjs.js!src/components/foo.svelte.css:
    Entrypoint mini-css-extract-plugin = *
       2 modules
```

With a bit of debugging, you can see it finds these resources for `bar.js`:
```
[ '/example/src/components/bar.svelte',
  '/example/node_modules/svelte/internal/index.mjs',
  '/example/src/components/bar.svelte.css' ]
```
And for `foo.js`:
```
[ '/example/src/components/foo.svelte.css' ]
```

build-dev works fine, finding these resources for the two entries:
```
[ '/example/src/bar.js' ]
[ '/example/src/foo.js' ]
```

**Comment by fqborges** [comment]
fixed on 0.5.0
